src/js/utils/general.js

```diff
@@ -85,65 +69,37 @@ export function uuid() {
   return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, (c) => {
     const r = (d + Math.random() * 16) % 16 | 0;
     d = Math.floor(d / 16);
-    return (c == 'x' ? r : (r & 0x3 | 0x8)).toString(16);
+    return (c == 'x' ? r : (r & 0x3) | 0x8).toString(16);
   });
 }
 
 /**
- * Gets the `Tether` options from a set of base `attachTo` options
+ * Gets the `Popper` options from a set of base `attachTo` options
  * @param attachToOptions
  * @param {Step} step The step instance
  * @return {Object}
  * @private
  */
-export function getTetherOptions(attachToOptions, step) {
-  let tetherOptions = {
-    classPrefix: 'shepherd',
-    constraints: [
-      {
-        to: 'scrollParent',
-        attachment: 'together',
-        pin: ['left', 'right', 'top']
-      },
-      {
-        to: 'window',
-        attachment: 'together'
-      }
-    ]
-  };
+export function getPopperOptions(attachToOptions, step) {
+  let popperOptions = {};
   let target = document.body;
 
   if (!attachToOptions.element || !attachToOptions.on) {
-    tetherOptions.attachment = 'middle center';
-    tetherOptions.targetModifier = 'visible';
+    // TODO add back centered popper hacks
+    // tetherOptions.attachment = 'middle center';
   } else {
-    tetherOptions.attachment = ATTACHMENT[attachToOptions.on] || ATTACHMENT.right;
+    popperOptions.placement = attachToOptions.on || 'right';
     target = attachToOptions.element;
   }
 
-  tetherOptions.element = step.el;
-  tetherOptions.target = target;
-
-  if (step.options.tetherOptions) {
-    if (step.options.tetherOptions.constraints) {
-      tetherOptions.constraints = step.options.tetherOptions.constraints;
-    }
-
-    tetherOptions.classes = {
-      ...tetherOptions.classes,
-      ...step.options.tetherOptions.classes
-    };
-
-    tetherOptions.optimizations = {
-      ...tetherOptions.optimizations,
-      ...step.options.tetherOptions.optimizations
-    };
+  if (step.options.popperOptions) {
+    // TODO add popper modifiers support etc
 
-    tetherOptions = {
-      ...tetherOptions,
-      ...step.options.tetherOptions
+    popperOptions = {
```

Comment:
You may need a smarter merge algorithm here if you want to allow modifiers options to be merged with your default